code/data_structures/src/stack/balanced_expression/balanced_expression.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <stack>
+
+using namespace std;
+
+bool isOpen(char c)
+{
+    if(c == '(' || c == '{' || c == '[')
+        return true;
+    return false;
+}
+
+bool isBalanced(char popped, char now)
+{
+    if(popped == '(' && now ==')')
+        return true;
+    if(popped == '[' && now ==']')
+        return true;
+    if(popped == '{' && now =='}')
+        return true;
+    return false;
+}
+
+int main()
+{
+    /* The code takes a string str as input from stdin and prints "YES" 
+    if str is a balanced expression else it prints "NO".*/
+        stack<char> s;
+        string str;
+        cin>>str;
+        int no = 0;
+        for(int i = 0; i < str.length(); i++)
```

Comment:
I think a range based for loop might be clearer, unless `i` is used somewhere in the loop.

Reply:
As `i` is being used inside the loop, a range based loop will lead in more execution time or might be insufficient if `str` is quite long. Plus using `str.length()` makes viewer exactly understand the code.